src/theory/arith/rewrites

```diff
@@ -51,3 +62,9 @@
 (define-rule* arith-plus-cancel1 ((t ? :list) (x ?) (s ? :list) (r ? :list)) (+ t x s (* (- 1) x) r) (+ t s r))
 (define-rule* arith-plus-cancel2 ((t ? :list) (x ?) (s ? :list) (r ? :list)) (+ t (* (- 1) x) s x r) (+ t s r))
 
+(define-rule arith-abs-elim ((x ?)) (abs x) (ite (< x 0) (- x) x))
+
+(define-rule arith-to-real-elim ((x Real)) (to_real x) x)
+(define-rule arith-to-int-elim-to-real ((x ?)) (to_int (to_real x)) (to_int x))
+(define-rule arith-div-elim-to-real1 ((x ?) (y ?)) (/ (to_real x) y) (/ x y))
+(define-rule arith-div-elim-to-real2 ((x ?) (y ?)) (/ x (to_real y)) (/ x y))
```

Comment:
Worried about these ones: are you sure it's safe to use these rules without re-introducing subtypes?